### PR TITLE
Stop Badger spamming slack channels

### PR DESCRIPTION
**Pull request by MuriloDalRi**
This bot used to only post to one channel but since it's been posting to the app owner's channels it's become too spammy and noisy.

Case in point:
Screenshot 2022-12-20 at 11 52 27

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -89,6 +85,6 @@ task :run do
       next
     end
 
-    HTTP.post(ENV.fetch("BADGER_SLACK_WEBHOOK_URL"), body: JSON.dump(message_payload))
+    HTTP.post(ENV.fetch("BADGER_SLACK_WEBHOOK_URL"), body: JSON.dump(message_payload)) unless message.nil?
```
Comment (ChrisBAshton) [inline Rakefile]: To make this a little easier to follow, I'd move the `nil` check onto the `message_payload` hash itself, i.e.

```ruby
message_payload = nil
if messages.any?
  message_payload = {
    ...
  }
end

HTTP.post(...) unless message_payload.nil?
```

Has the nice side effect of doing away with the temporary `message` variable that no longer has any ternary logic on it.

**Review by PeterHattyar — APPROVED** [review]

File: Rakefile (hunk 1 above)
Comment (MuriloDalRi) [reply Rakefile]: I've changed it now

File: Rakefile [hunk 2]
```diff
@@ -56,19 +56,17 @@ task :run do
...
-              end
-
-    message_payload = {
-      username: "Badger",
-      icon_emoji: ":badger:",
-      text: message,
-      mrkdwn: true,
-      channel: team,
-    }
+    message = "Hello :paw_prints:, this is your <https://github.com/alphagov/govuk-deploy-lag-badger|regular badgering to deploy>!\n\n#{messages.join("\n")}" if messages.any?
```
Comment (ChrisBAshton) [inline Rakefile]: You can now move this string directly into line 65, so that the only conditional check happening is on line 61.

Comment (MuriloDalRi) [reply Rakefile]: Ah yes, too eager to get rid of this that I didn't notice 😄

**Pull request merged**